src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const currentState = { ...state };
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(currentState, obj.extraData);
```

Comment:
you mutate one initial cloned object all the time
It is better to create a copy on each iteration

Reply:
Thanks for advice!

Reply:
I don't understand.

Reply:
You created a copy before the loop
and modify it on every action
But it is better to create a copy for each action 
I'll show you tomorrow